pkg/eestream/rs_test.go

```diff
@@ -32,3 +37,83 @@ func TestRS(t *testing.T) {
 		t.Fatalf("rs encode/decode failed")
 	}
 }
+
+func TestRSErrors(t *testing.T) {
+	for i, tt := range []struct {
+		dataSize    int
+		blockSize   int
+		required    int
+		total       int
+		problematic int
+		fail        bool
+	}{
+		{4 * 1024, 1024, 1, 1, 0, false},
+		{4 * 1024, 1024, 1, 1, 1, true},
+		{4 * 1024, 1024, 1, 2, 0, false},
+		{4 * 1024, 1024, 1, 2, 1, false},
+		{4 * 1024, 1024, 1, 2, 2, true},
+		{4 * 1024, 1024, 2, 4, 0, false},
+		{4 * 1024, 1024, 2, 4, 1, false},
+		{4 * 1024, 1024, 2, 4, 2, false},
+		{4 * 1024, 1024, 2, 4, 3, true},
+		{4 * 1024, 1024, 2, 4, 4, true},
+		{6 * 1024, 1024, 3, 7, 0, false},
+		{6 * 1024, 1024, 3, 7, 1, false},
+		{6 * 1024, 1024, 3, 7, 2, false},
+		{6 * 1024, 1024, 3, 7, 3, false},
+		{6 * 1024, 1024, 3, 7, 4, false},
+		{6 * 1024, 1024, 3, 7, 5, true},
+		{6 * 1024, 1024, 3, 7, 6, true},
+		{6 * 1024, 1024, 3, 7, 7, true},
+	} {
+		errTag := fmt.Sprintf("Test case #%d", i)
+		data := randData(tt.dataSize)
+		fc, err := infectious.NewFEC(tt.required, tt.total)
+		if !assert.NoError(t, err, errTag) {
+			continue
+		}
+		rs := NewRSScheme(fc, tt.blockSize)
+		readers := EncodeReader(bytes.NewReader(data), rs)
+		// read all readers in []byte buffers to avoid deadlock if later
+		// we don't read in parallel from all of them
+		pieces, err := readAll(readers)
+		if !assert.NoError(t, err, errTag) {
```

Comment:
nit - https://godoc.org/github.com/stretchr/testify/assert#Error

Reply:
`if !assert.NoError(...) { continue}` is not the same as `if assert.Error(...) { continue }`.

Here I assert that there is no error, but if the assertion fails the test case should fail and continue with the next one immediately.

If I replace it with `if assert.Error(...) { continue }` then the tests will fail because there is no error, which is wrong.